### PR TITLE
performance headlines / landing page

### PR DESCRIPTION
**Pull request by ctz**
This PR adds a little visual display of performance "headlines" to https://rustls.dev/perf/ which looks like:

image

These are just server TLS1.3 handshake performance, plus TLS1.3 AES-256-GCM transfer rates, which are think is a reasonable 

Most of this PR is made up of an extremely "clever" macro, which expands to an SVG I drew in inkscape and sets the dial transforms based on its inputs. I suggest this SVG is treated as write-only :)

### CODE REVIEW

**Review by djc — APPROVED** [review]
Looks fun!

I think the numerical differences (-47%) are a bit confusing? I think a factor ("1.9x slower") would be easier to grok.

File: website/templates/reports.html [hunk 1]
```diff
@@ -1,6 +1,119 @@
...
+              <h2>{{ page.extra.transfer.recv.boringssl | round }}
+                {{ macros::compare(target=page.extra.transfer.recv.boringssl, base=page.extra.transfer.recv.rustls) }}
+                <footer>BoringSSL</footer>
+              </h2>
+            </div>
+          </div>
+        </div>
+
+      </div>
+      <!-- {{ page.extra.version.rustls }} vs {{ page.extra.version.openssl }} vs {{ page.extra.version.boringssl }} -->
```
Comment (cpu) [inline website/templates/reports.html:107]: Should this be left in?

**Review by cpu — APPROVED** [review]
Nice :+1: Glad to see some of the functionality of the site generator being helpful sooner than later.

**Comment by ctz** [comment]
> Nice 👍 Glad to see some of the functionality of the site generator being helpful sooner than later.

I think just for autoreload it has been worth it so far

**Comment by djc** [comment]
Could you update the screenshot?

I wonder if we could get this to do deploys for PRs...

**Comment by ctz** [comment]
> Could you update the screenshot?

done

> I wonder if we could get this to do deploys for PRs...

apparently there is a thing called "GitHub Pages preview" but "This feature is only in alpha currently and is not available to the public!" so maybe this will become easy soon?

alternatively we could deploy prs to a throw-away netlify account or similar

**Review by djc — APPROVED** [review]

**Pull request merged**